Pull request

delioncourts opened the pull request.

delioncourts committed:
upd readme

delioncourts committed:
add CRA

delioncourts committed:
add structure, normalize

delioncourts committed:
add fonts

delioncourts committed:
add images

delioncourts committed:
upd Header, Not Found, Promo

delioncourts committed:
delete navtab

delioncourts committed:
upd App, Main, Promo

delioncourts committed:
upd aputme, app, header and headerlogo

delioncourts committed:
upd AboutProject, App, export default

delioncourts committed:
upd AboutProjects, Footer

delioncourts committed:
upd Techs

delioncourts committed:
upd AboutMe, exports

delioncourts committed:
upd App, Login, Portfolio, Profile

delioncourts committed:
upd AboutMe, AboutProject, Checkbox, Error, Footer, hamburgerMenu

delioncourts committed:
upd Promo, About

delioncourts committed:
upd AboutMe, checkbox, Techs, Promo

delioncourts committed:
upd profile

delioncourts committed:
upd structure Movies, Register, Login

delioncourts committed:
upd login, movies

delioncourts committed:
fix Promo css

delioncourts committed:
upd CSS Main

delioncourts committed:
upd Login and Register

delioncourts committed:
upd PageNotFound, Register and Profile

delioncourts committed:
upd Registwer, fix headerLogo, Login

delioncourts committed:
upd Navigation, Hamburger, Profile

delioncourts committed:
upd checkbox. searchform

delioncourts committed:
upd movies, opacity and transition

delioncourts committed:
upd hamburger transition

coolswood commented:
# Резюме по работе:
## Верстка и JSX

### Критерии, влияющие на работоспособность
- [ ] <!--12--> Все блоки и элементы из макета сверстаны. Размеры и расположение не отличаются от макета на более, чем на 30px на десктопной версии и на 10px на остальных размерах экрана, для которых подготовлен макет. Отличия в размерах не должны визуально противоречить макету и выбиваться из заложенной в макет сетки, в этом случае требуется более строгое соответствие размерам макета.
`На расширении 320рх цвет фонового блока techs не соответствует макету. Обратите внимание что цвет в макете меняется`

- [x] <!--10.55--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
- [x] <!--9.55--> Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вклxaадке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить.
- [x] <!--5.55--> Именование классов сделаны по БЭМ.
- [x] <!--8.55--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях.
- [ ] <!--3.38--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфоxм, список — списком); структура DOM-дерева состоит не только из контейнеров div.
`Кнопкам всегда нужно указывать тип. Потому что по-умолчанию это submit и такая кнопка при клике может отправить форму.`

- [x] <!--3.38--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
- [x] <!--3.38--> Каркас макета реализован на Flex layout или Grid layout;
- [x] <!--3.38--> Инфраструктурные файлы проекта созданы через CRA.
- [x] <!--3.38--> Работа модальных окон настроена.
- [x] Разметка портирована в JSX:
	- [x] <!--3.38--> разметка закчлючена в ( ), лишние пустые <div> для этого не используются;
	- [x] <!--3.38--> разметка вынесена в соответствующие ей компоненты.
- [x] В проекте есть:
	- [x] <!--3.38--> директория images/ с картинками,
	- [x] <!--3.38--> директория components/ с JS и CSS-файлами компонентов,
	- [x] <!--3.38--> директория fonts/ со шрифтами.

### Хорошие практики
- [x] <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете. 
- [x] <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- [x] <!--2.14--> Шрифты подключены через @font-face.
- [x] <!--2.14--> Иконки экспортированы в формате SVG.
- [x] <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение.
- [ ] <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля.
`У формы поиска фильмов нет свойства required`
- [x] <!--2.14--> reset.css запрещен.

### Рекомендации
- [x] <!--1.68--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- [x] <!--1.67--> Для изображений задан атрибут alt с подходящий значением.
- [x] <!--1.67--> Растровые и векторные изображения оптимизированы;

### Рекомендации
`Неправильно задан lang`

## Количество баллов: 82

delioncourts committed:
fix after review: lang, techs, button

[X] изменила язык с eng на ru;
[X] добавила тип кнопок и aria-label;
[X] изменила цвет фона у Techs на разрешении 320px.
